src/kvstore/comm.h

```diff
@@ -170,34 +170,36 @@ class CommCPU : public Comm {
     }
   }
 
-  // TODO(haibin) support broadcast row_sparse on GPU
   void BroadcastRowSparse(int key, const NDArray& src,
                           const std::vector<std::pair<NDArray*, NDArray>>& dst,
                           const bool use_copy,
                           const int priority) override {
     using namespace mshadow;
-    auto size = dst.size();
-    for (size_t i = 0; i < size; i++) {
-      auto out = dst[i].first;
-      auto row_id = dst[i].second;
+    for (size_t i = 0; i < dst.size(); ++i) {
```

Comment:
In most std implementations, vector::size is inlined. In terms of micro-optimization vs. readability and maintainability, I prefer the latter one unless that kind of optimization generates noticeable improvements. Falling into the habit of caching vector size is dangerous in practice. In addition, creating a local variable doesn't guarantee a performance improvement. If your register is full, reading the local variable from stack also has extra overhead. I agree that we may evaluate this kind of micro-optimization in the cases of dealing tight loops. However, the current loop is not trivial.

I also tried two approaches using the benchmark script in this PR and didn't find any noticeable difference between them.

Here are some good discussions:
https://stackoverflow.com/a/2753637
https://stackoverflow.com/a/8613000